### PR TITLE
New cross compilers

### PR DESCRIPTION
**Pull request by astralien3000**
Hello !

I a making this PR with the aim managing microcontrollers with conan.
The first thing is to add some cross-compilers.

EDIT: 
This is the first PR related to #86, but not the last !

### CODE REVIEW

File: conans/client/conf/__init__.py [hunk 1]
```diff
@@ -25,6 +25,12 @@
     apple-clang:
         version: ["5.0", "5.1", "6.0", "6.1", "7.0", "7.3"]
         libcxx: [libstdc++, libc++]
+    arm-none-eabi-gcc:
+        version: ["5.2"]
+        libcxx: [None]
```
Comment (memsharded) [inline conans/client/conf/__init__.py]: If there is only one setting and it is None, we can omit this setting at the moment, and add it later if necessary. The value "None" in the future is allowed to not change the hash of existing package binaries, so it shouldnt be a problem.

**Comment by memsharded** [comment]
Looks good to me (except I would omit the libcxx by now), any feedback @weatherhead99 and @lasote?

**Comment by astralien3000** [comment]
OK for libcxx.
About the version, I only added the ones that are on my computer, is it enough ?

**Comment by lasote** [comment]
I'm not sure about a compiler called "arm-none-eabi-gcc" and not a "gcc" compiler with an "armxxx" architecture or something like that. Do you have any public project to see if I get some background and think about it? Are you building things for android or ...?

**Comment by weatherhead99** [comment]
@lasote arm-none-eabi-gcc or gcc-arm-none-eabi is the canonical name for the open source gcc based "bare metal" toolchain worked on by ARM. It can be found here:
https://launchpad.net/gcc-arm-embedded/

**Comment by astralien3000** [comment]
+1
Cross-compilers based on gcc are not the same compiler as gcc. It is not an option you can pass to the basic gcc compiler. Most are based on older versions, and is is not the same binary.

I know a lot of project that uses this compiler : 
 - [Aversive++](https://github.com/AversivePlusPlus/toolchain-stm32/blob/f269765b021a829e35aa7af71c9366d786984c63/stm32.cmake)
 - [RIOT](https://github.com/RIOT-OS/RIOT/wiki/Family%3A-ARM)
 - [xpcc](https://github.com/roboterclubaachen/xpcc/blob/20e4aab3a670c27ac88111c7a51881e111972e8d/scons/site_tools/arm.py)

**Comment by lasote** [comment]
Thanks for the information. So, in the conanfile that you have... which settings are specified? just the compiler?

**Comment by astralien3000** [comment]
I have made a first [conan project able to use the compiler](https://github.com/AversivePlusPlus/template-project/tree/dev-conan). But is uses more the "mcu" (MicroController Unit) setting, since it uses a CMake toolchain to select the compiler.

The command I used is : 
```bash
conan install .. -s compiler=arm-none-eabi-gcc -s compiler.version=5.2 -s compiler.libcxx=None -s target=stm32f407vg
```

**Comment by lasote** [comment]
So the:

```
class AversivePlusPlusProjectConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch", "mcu"
```
could be simplified as:

```
class AversivePlusPlusProjectConan(ConanFile):
    settings = "compiler", "build_type", "mcu"
```

Because target will always ARM without operating system, right?
That's why I'm not sure about it... the compiler now contains the arch and the so somehow...

**Comment by astralien3000** [comment]
Yes, it could. It is not compulsory to be ARM (it could be AVR, or something).
That is what we were talking about in #86, with cross compilation, the "mcu" contains the information of the arch, and prevent to use some compilers (avr-gcc cannot be used on ARM targets), and may not use the OS setting.

**Comment by astralien3000** [comment]
Well, maybe it is better to have a complete solution for cross compilation before making a pull request ? You can close if you agree, and I could come back when I have a complete solution.
Or would you rather go step by step ?

**Comment by lasote** [comment]
@astralien3000 Yes, that my doubt, There are some requests of cross compilation support and I doubt about the better approach. @memsharded what do you think? maybe we would need to stop and think about it quietly?

**Pull request closed without merging**